### PR TITLE
Memory efficient (mid-)long term cpu/mem gauge storage

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

File: metrics/sinks/metric/metric_sink.go [hunk 1]
```diff
@@ -37,7 +37,13 @@ type MetricSink struct {
 	// Stores full DataBatch with all metrics and labels
 	shortStore []*core.DataBatch
 	// Stores timmed DataBatches with selected metrics and no labels
-	longStore []*core.DataBatch
+	longStore []*multimetricStore
+}
+
+type int64Store map[string]int64
+type multimetricStore struct {
+	timestamp time.Time
```
Comment (fgrzadkowski) [inline metrics/sinks/metric/metric_sink.go]: What does this timestamp represent? Please add comments.

File: metrics/sinks/metric/metric_sink.go [hunk 2]
```diff
@@ -45,6 +51,25 @@ type TimestampedMetricValue struct {
 	Timestamp time.Time
 }
 
+func buildMultimetricStore(metrics []string, batch *core.DataBatch) *multimetricStore {
+	store := multimetricStore{
+		timestamp: batch.Timestamp,
+		store:     make(map[string]int64Store, len(metrics)),
+	}
+	for _, metric := range metrics {
+		store.store[metric] = make(map[string]int64, len(batch.MetricSets))
```
Comment (fgrzadkowski) [inline metrics/sinks/metric/metric_sink.go]: Can you use `int64Store` here?

File: metrics/sinks/metric/metric_sink.go (hunk 1 above)
Comment (mwielgus) [inline metrics/sinks/metric/metric_sink.go]: Added

File: metrics/sinks/metric/metric_sink.go (hunk 2 above)
Comment (mwielgus) [inline metrics/sinks/metric/metric_sink.go]: Done.

File: metrics/sinks/metric/metric_sink.go [hunk 3]
```diff
@@ -26,18 +26,29 @@ import (
 // * metrics that need to be stored for longer time (15 min, 1 hour).
 // The user of this struct needs to decide what are the long-stored metrics uprfront.
 type MetricSink struct {
-	// Request can come from other threads
+	// Request can come from other threads.
 	lock sync.Mutex
 
-	// list of metrics that will be stored for up to X seconds
+	// list of metrics that will be stored for up to X seconds.
```
Comment (fgrzadkowski) [inline metrics/sinks/metric/metric_sink.go]: Start with capital letter.

File: metrics/sinks/metric/metric_sink.go [hunk 4]
```diff
@@ -26,18 +26,29 @@ import (
...
+	// Memory-efficient long/mid term storage for metrics.
+	longStore []*multimetricStore
+}
+
+// Stores values of a single metrics for different MetricSets.
+// Assumes that the user knows what the metric is.
+type int64Store map[string]int64
+
+type multimetricStore struct {
+	// Timestamp of the batch from which the metrics were takes.
```
Comment (fgrzadkowski) [inline metrics/sinks/metric/metric_sink.go]: s/takes/taken

File: metrics/sinks/metric/metric_sink.go (hunk 3 above)
Comment (mwielgus) [inline metrics/sinks/metric/metric_sink.go]: Done

**Pull request merged**